bacs/src/main/java/com/adyen/checkout/bacs/BacsProvider.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 12/4/2022.
+ */
+
+package com.adyen.checkout.bacs
+
+import android.os.Bundle
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.savedstate.SavedStateRegistryOwner
+import com.adyen.checkout.components.PaymentComponentProvider
+import com.adyen.checkout.components.base.GenericPaymentMethodDelegate
+import com.adyen.checkout.components.base.lifecycle.viewModelFactory
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+
+class BacsProvider : PaymentComponentProvider<BacsDirectDebitComponent, BacsDirectDebitConfiguration> {
```

Comment:
Tbh I actually like the `...ComponentProvider` suffix 😅, because it exactly tells what it provides. Alternatively, the provider could be a sub class of the component, so you get `BacsComponent.Provider`

Reply:
That would be interesting but I don't think I can do it in another file can I?

Reply:
No, unfortunately not. AFAIK there is nothing like extensions classes :P

Reply:
I'm adding the suffix again.